qudipy/potential/load_data.py

```diff
@@ -246,4 +247,55 @@ def load_potentials(ctrl_vals, ctrl_names, f_type='pot', f_dir=None,
     all_files['potentials'] = pots_array
     
     
-    return all_files
\ No newline at end of file
+    return all_files
+
+def analytical_potential(ctrl_vals, ctrl_names, function, x_range, y_range):
+    '''
+    Allows one to create an analytical potential in the xy-plane from a function
+    Parameters
+    ----------
+    ctrl_vals : list of list of floats
+        List of relevant control values for the files to load.  The first list
+        index corresponds to the ith control variable and the second list
+        index correspond to the ith value for that control variable.
+    ctrl_names : list of strings
+        List of each ctrl variable name. Must be the same length as ctrl_vals 
+        first dimension.
+    function : callable
+        Function which defines the 2D potential you wish to map out. function
+        must take the list ctrl_vals as the first argument and a GridParameters 
+        object as the second argument. Refer to tutorial for an explicit example
+    x_range : List or Numpy array
+        grid points along x where you want to potential to be calculated
+    y_range : List or Numpy array
+        grid points along y where you want to potential to be calculated
+
+    Returns
+    -------
+    analytical_potential : dictionary
+        Contains all the information necessary to create an interpolator from the 
+        analytical potential.
+    '''
+
+    # From the x and y ranges provided, create the GridParameters
+    gparams = GridParameters(x_range, y_range)
+
+    # Go over all products of control values and 
+    cval_array = []
+    pots_array = []
+    for _, curr_cvals in enumerate(product(*ctrl_vals)):
```

Comment:
```suggestion
    for curr_cvals in product(*ctrl_vals):
```

You are not using the 1st output of `enumerate`; therefore, you can slightly simplify this line